### PR TITLE
KF-4262 contributing

### PR DESCRIPTION
**Pull request by ColmBhandal**
Added contributing files for all KFP charms.

Content is generic.

### CODE REVIEW

File: charms/kfp-api/contributing_inputs.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+  ## Updating the charm for new versions of the workload
+
+  To upgrade the source and resources of this charm, you must:
+  
+  1. Bump the `oci-image` in `metadata.yaml`
+  1. Update the charm source for any changes, such as:
+      - YAML manifests in `src/` and/or any Kubernetes resource in `pod_spec`
+      - New or changed configurations passed to pebble workloads or through `pod.set_spec`
+  1. Ensure integration and unit tests are passing; fix/adapt them otherwise
```
Comment (phoevos) [inline charms/kfp-api/contributing_inputs.yaml:19]: Let's add a new line at the end of the file, please.

File: charms/kfp-persistence/contributing_inputs.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+  ## Updating the charm for new versions of the workload
+
+  To upgrade the source and resources of this charm, you must:
+  
+  1. Bump the `oci-image` in `metadata.yaml`
+  1. Update the charm source for any changes, such as:
+      - YAML manifests in `src/` and/or any Kubernetes resource in `pod_spec`
+      - New or changed configurations passed to pebble workloads or through `pod.set_spec`
+  1. Ensure integration and unit tests are passing; fix/adapt them otherwise
```
Comment (phoevos) [inline charms/kfp-persistence/contributing_inputs.yaml:19]: Let's add a new line at the end of the file, please.

File: charms/kfp-profile-controller/contributing_inputs.yaml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+  ## Updating the charm for new versions of the workload
+
+  To upgrade the source and resources of this charm, you must:
+  
+  1. Bump the `oci-image` in `metadata.yaml`
+  1. Update the charm source for any changes, such as:
+      - YAML manifests in `src/` and/or any Kubernetes resource in `pod_spec`
+      - New or changed configurations passed to pebble workloads or through `pod.set_spec`
+  1. Ensure integration and unit tests are passing; fix/adapt them otherwise
```
Comment (phoevos) [inline charms/kfp-profile-controller/contributing_inputs.yaml:19]: Let's add a new line at the end of the file, please.

File: charms/kfp-schedwf/contributing_inputs.yaml [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+  ## Updating the charm for new versions of the workload
+
+  To upgrade the source and resources of this charm, you must:
+  
+  1. Bump the `oci-image` in `metadata.yaml`
+  1. Update the charm source for any changes, such as:
+      - YAML manifests in `src/` and/or any Kubernetes resource in `pod_spec`
+      - New or changed configurations passed to pebble workloads or through `pod.set_spec`
+  1. Ensure integration and unit tests are passing; fix/adapt them otherwise
```
Comment (phoevos) [inline charms/kfp-schedwf/contributing_inputs.yaml:19]: Let's add a new line at the end of the file, please.

File: charms/kfp-ui/contributing_inputs.yaml [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+  ## Updating the charm for new versions of the workload
+
+  To upgrade the source and resources of this charm, you must:
+  
+  1. Bump the `oci-image` in `metadata.yaml`
+  1. Update the charm source for any changes, such as:
+      - YAML manifests in `src/` and/or any Kubernetes resource in `pod_spec`
+      - New or changed configurations passed to pebble workloads or through `pod.set_spec`
+  1. Ensure integration and unit tests are passing; fix/adapt them otherwise
```
Comment (phoevos) [inline charms/kfp-ui/contributing_inputs.yaml:19]: Let's add a new line at the end of the file, please.

File: charms/kfp-viewer/contributing_inputs.yaml [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+
+  ## Updating the charm for new versions of the workload
+
+  To upgrade the source and resources of this charm, you must:
+  
+  1. Bump the `oci-image` in `metadata.yaml`
+  1. Update the charm source for any changes, such as:
+      - YAML manifests in `src/` and/or any Kubernetes resource in `pod_spec`
+      - New or changed configurations passed to pebble workloads or through `pod.set_spec`
+  1. Ensure integration and unit tests are passing; fix/adapt them otherwise
```
Comment (phoevos) [inline charms/kfp-viewer/contributing_inputs.yaml:19]: Let's add a new line at the end of the file, please.

File: charms/kfp-viz/contributing_inputs.yaml [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+
+  ## Updating the charm for new versions of the workload
+
+  To upgrade the source and resources of this charm, you must:
+  
+  1. Bump the `oci-image` in `metadata.yaml`
+  1. Update the charm source for any changes, such as:
+      - YAML manifests in `src/` and/or any Kubernetes resource in `pod_spec`
+      - New or changed configurations passed to pebble workloads or through `pod.set_spec`
+  1. Ensure integration and unit tests are passing; fix/adapt them otherwise
```
Comment (phoevos) [inline charms/kfp-viz/contributing_inputs.yaml:19]: Let's add a new line at the end of the file, please.

File: charms/kfp-viewer/contributing_inputs.yaml [hunk 8]
```diff
@@ -0,0 +1,19 @@
+charm: "kfp-viewer"
+tox_example_dev_env: "lint"
+os_base: "ubuntu-20.04"
+deploy_opts: |-
+  \
+    --resource oci-image=$(yq '.resources."oci-image"."upstream-source"' metadata.yaml)
```
Comment (phoevos) [inline charms/kfp-viewer/contributing_inputs.yaml:6]: See: https://github.com/canonical/kfp-operators/blob/38fd0f0d8ae5a6195930add44543700986bfc891/charms/kfp-viewer/metadata.yaml#L11-L15
```suggestion
    --resource oci-image=$(yq '.resources."kfp-viewer-image"."upstream-source"' metadata.yaml)
```

Comment (phoevos) [reply charms/kfp-viewer/contributing_inputs.yaml:6]: Then please rerender `contributing.md`.

File: charms/kfp-ui/contributing_inputs.yaml [hunk 9]
```diff
@@ -0,0 +1,19 @@
+charm: "kfp-ui"
+tox_example_dev_env: "lint"
+os_base: "ubuntu-20.04"
+deploy_opts: |-
+  \
+    --resource oci-image=$(yq '.resources."oci-image"."upstream-source"' metadata.yaml)
```
Comment (phoevos) [inline charms/kfp-ui/contributing_inputs.yaml:6]: See: https://github.com/canonical/kfp-operators/blob/38fd0f0d8ae5a6195930add44543700986bfc891/charms/kfp-ui/metadata.yaml#L9-L13

```suggestion
    --resource oci-image=$(yq '.resources."ml-pipeline-ui"."upstream-source"' metadata.yaml)
```

Then please rerender `contributing.md`.

**Review by phoevos — CHANGES_REQUESTED** [review]
Thanks @ColmBhandal, some changes.
Note, that we should also rerender the `contributing.md` files based on the change introduced [here](https://github.com/canonical/kubeflow-ci/pull/107), in order to avoid a follow-up PR.

**Comment by ColmBhandal** [comment]
> Thanks @ColmBhandal, some changes. Note, that we should also rerender the `contributing.md` files based on the change introduced [here](https://github.com/canonical/kubeflow-ci/pull/107), in order to avoid a follow-up PR.

I'm working on fixing the automation. It should not result in a follow-up PR to `main`, but rather a PR targeting this branch which we can just merge. I just need to fix the issue that seems to be with permissions...

**Comment by ColmBhandal** [comment]
Closing this as it's too much effort. We have decided for now to go with the simpler option of just manually creating and maintaining contributing files.

**Pull request closed without merging**